### PR TITLE
Add find and delete on documentQuery, #144.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/generator/AbstractQueryGenerator.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        switch (type) {
+            case IS_EQUAL:
+                return this.generateIsEqual(criteria, parameters);
+            case AND:
+            case OR:
+                Assert.isTrue(criteria.getSubCriteria().size() == 2, "criteria should have two SubCriteria");
+
+                final String left = generateQueryBodyDfs(criteria.getSubCriteria().get(0), parameters);
+                final String string = generateQueryBodyDfs(criteria.getSubCriteria().get(1), parameters);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/generator/AbstractQueryGenerator.java]: `string` -> `right`?

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 2]
```diff
@@ -395,6 +400,56 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
...
 
+    public <T> List<T> find(@NonNull DocumentQuery query, Class<T> domainClass, @NonNull String collectionName) {
+        Assert.hasText(collectionName, "collection should not be null, empty or only whitespaces");
+
+        final FeedOptions feedOptions = new FeedOptions();
+        final DocumentCollection collection = getDocCollection(collectionName);
+        final Optional<String> partitionKeyName = getPartitionKeyField(domainClass);
+        final QuerySpecGenerator generator = new FindQueueSpecGenerator(domainClass);
+
+        feedOptions.setEnableCrossPartitionQuery(!query.getSubjectCriteria(partitionKeyName.orElse("")).isPresent());
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Move `!query.getSubjectCriteria(partitionKeyName.orElse("")).isPresent()` as a variable?

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/generator/AbstractQueryGenerator.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/generator/AbstractQueryGenerator.java]: yes, I mean right here, will refine this.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: OK, should be align with delete code.

**Pull request merged**